fail during linktest

```make
./gensymbol linktest  x86_64 _ 0 0 0 0 0 0 "" "" 1 0 1 1 1 1 > linktest.c
gcc -O2 -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.21\" -msse3 -mssse3 -msse4.1 -mavx -mavx2 -mavx2 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  -shared -o ../libopenblas_haswellp-r0.3.21.so \
-Wl,--whole-archive ../libopenblas_haswellp-r0.3.21.a -Wl,--no-whole-archive \
-Wl,-soname,libopenblas.so.0 -lm -lpthread -lgfortran -lm -lpthread -lgfortran
gcc -O2 -DSMALL_MATRIX_OPT -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -fPIC -DNO_AVX512 -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=56 -DMAX_PARALLEL_NUMBER=1 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.21\" -msse3 -mssse3 -msse4.1 -mavx -mavx2 -mavx2 -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  -w -o linktest linktest.c ../libopenblas_haswellp-r0.3.21.so -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5 -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5/../../../../lib64 -L/lib/../lib64 -L/usr/lib/../lib64 -L/es01/software/oneapi/vpl/2021.2.2/lib -L/es01/software/oneapi/tbb/2021.2.0/env/../lib/intel64/gcc4.8 -L/es01/software/oneapi/mpi/2021.2.0//libfabric/lib -L/es01/software/oneapi/mpi/2021.2.0//lib/release -L/es01/software/oneapi/mpi/2021.2.0//lib -L/es01/software/oneapi/mkl/latest/lib/intel64 -L/es01/software/oneapi/ipp/2021.2.0/lib/intel64 -L/es01/software/oneapi/ippcp/2021.2.0/lib/intel64 -L/es01/software/oneapi/ipp/2021.2.0/lib/intel64 -L/es01/software/oneapi/dnnl/2021.2.0/cpu_dpcpp_gpu_dpcpp/lib -L/es01/software/oneapi/dal/2021.2.0/lib/intel64 -L/es01/software/oneapi/compiler/2021.2.0/linux/compiler/lib/intel64_lin -L/es01/software/oneapi/compiler/2021.2.0/linux/lib -L/es01/software/oneapi/clck/2021.2.0/lib/intel64 -L/es01/software/oneapi/ccl/2021.2.0/lib/cpu_gpu_dpcpp -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5/../../..  -lgfortran -lm -lquadmath -lm -lc  && echo OK.
/tmp/cc1j7MgI.o: In function `main':
linktest.c:(.text.startup+0xf88): undefined reference to `slagge_'
linktest.c:(.text.startup+0xf8f): undefined reference to `slagsy_'
linktest.c:(.text.startup+0xf96): undefined reference to `slahilb_'
linktest.c:(.text.startup+0xf9d): undefined reference to `slakf2_'
linktest.c:(.text.startup+0xfa4): undefined reference to `slaran_'
linktest.c:(.text.startup+0xfab): undefined reference to `slarge_'
linktest.c:(.text.startup+0xfb2): undefined reference to `slarnd_'
linktest.c:(.text.startup+0xfb9): undefined reference to `slaror_'
linktest.c:(.text.startup+0xfc0): undefined reference to `slarot_'
linktest.c:(.text.startup+0xfc7): undefined reference to `slatm1_'
linktest.c:(.text.startup+0xfce): undefined reference to `slatm2_'
linktest.c:(.text.startup+0xfd5): undefined reference to `slatm3_'
linktest.c:(.text.startup+0xfdc): undefined reference to `slatm5_'
linktest.c:(.text.startup+0xfe3): undefined reference to `slatm6_'
linktest.c:(.text.startup+0xfea): undefined reference to `slatm7_'
linktest.c:(.text.startup+0xff1): undefined reference to `slatme_'
linktest.c:(.text.startup+0xff8): undefined reference to `slatmr_'
linktest.c:(.text.startup+0xfff): undefined reference to `slatms_'
linktest.c:(.text.startup+0x1006): undefined reference to `slatmt_'
linktest.c:(.text.startup+0x1b90): undefined reference to `dlagge_'
linktest.c:(.text.startup+0x1b97): undefined reference to `dlagsy_'
linktest.c:(.text.startup+0x1b9e): undefined reference to `dlahilb_'
linktest.c:(.text.startup+0x1ba5): undefined reference to `dlakf2_'
linktest.c:(.text.startup+0x1bac): undefined reference to `dlaran_'
linktest.c:(.text.startup+0x1bb3): undefined reference to `dlarge_'
linktest.c:(.text.startup+0x1bba): undefined reference to `dlarnd_'
linktest.c:(.text.startup+0x1bc1): undefined reference to `dlaror_'
linktest.c:(.text.startup+0x1bc8): undefined reference to `dlarot_'
linktest.c:(.text.startup+0x1bcf): undefined reference to `dlatm1_'
linktest.c:(.text.startup+0x1bd6): undefined reference to `dlatm2_'
linktest.c:(.text.startup+0x1bdd): undefined reference to `dlatm3_'
linktest.c:(.text.startup+0x1be4): undefined reference to `dlatm5_'
linktest.c:(.text.startup+0x1beb): undefined reference to `dlatm6_'
linktest.c:(.text.startup+0x1bf2): undefined reference to `dlatm7_'
linktest.c:(.text.startup+0x1bf9): undefined reference to `dlatme_'
linktest.c:(.text.startup+0x1c00): undefined reference to `dlatmr_'
linktest.c:(.text.startup+0x1c07): undefined reference to `dlatms_'
linktest.c:(.text.startup+0x1c0e): undefined reference to `dlatmt_'
linktest.c:(.text.startup+0x28b7): undefined reference to `clagge_'
linktest.c:(.text.startup+0x28be): undefined reference to `claghe_'
linktest.c:(.text.startup+0x28c5): undefined reference to `clagsy_'
linktest.c:(.text.startup+0x28cc): undefined reference to `clahilb_'
linktest.c:(.text.startup+0x28d3): undefined reference to `clakf2_'
linktest.c:(.text.startup+0x28da): undefined reference to `clarge_'
linktest.c:(.text.startup+0x28e1): undefined reference to `clarnd_'
linktest.c:(.text.startup+0x28e8): undefined reference to `claror_'
linktest.c:(.text.startup+0x28ef): undefined reference to `clarot_'
linktest.c:(.text.startup+0x28f6): undefined reference to `clatm1_'
linktest.c:(.text.startup+0x28fd): undefined reference to `clatm2_'
linktest.c:(.text.startup+0x2904): undefined reference to `clatm3_'
linktest.c:(.text.startup+0x290b): undefined reference to `clatm5_'
linktest.c:(.text.startup+0x2912): undefined reference to `clatm6_'
linktest.c:(.text.startup+0x2919): undefined reference to `clatme_'
linktest.c:(.text.startup+0x2920): undefined reference to `clatmr_'
linktest.c:(.text.startup+0x2927): undefined reference to `clatms_'
linktest.c:(.text.startup+0x292e): undefined reference to `clatmt_'
linktest.c:(.text.startup+0x33fb): undefined reference to `zlagge_'
linktest.c:(.text.startup+0x3402): undefined reference to `zlaghe_'
linktest.c:(.text.startup+0x3409): undefined reference to `zlagsy_'
linktest.c:(.text.startup+0x3410): undefined reference to `zlahilb_'
linktest.c:(.text.startup+0x3417): undefined reference to `zlakf2_'
linktest.c:(.text.startup+0x341e): undefined reference to `zlarge_'
linktest.c:(.text.startup+0x3425): undefined reference to `zlarnd_'
linktest.c:(.text.startup+0x342c): undefined reference to `zlaror_'
linktest.c:(.text.startup+0x3433): undefined reference to `zlarot_'
linktest.c:(.text.startup+0x343a): undefined reference to `zlatm1_'
linktest.c:(.text.startup+0x3441): undefined reference to `zlatm2_'
linktest.c:(.text.startup+0x3448): undefined reference to `zlatm3_'
linktest.c:(.text.startup+0x344f): undefined reference to `zlatm5_'
linktest.c:(.text.startup+0x3456): undefined reference to `zlatm6_'
linktest.c:(.text.startup+0x345d): undefined reference to `zlatme_'
linktest.c:(.text.startup+0x3464): undefined reference to `zlatmr_'
linktest.c:(.text.startup+0x346b): undefined reference to `zlatms_'
linktest.c:(.text.startup+0x3472): undefined reference to `zlatmt_'
collect2: error: ld returned 1 exit status
make[1]: *** [../libopenblas_haswellp-r0.3.21.so] Error 1
make[1]: Leaving directory `/es01/home/liyl/OpenBLAS-0.3.21/exports'
make: *** [shared] Error 2
```

with gcc version 4.8.5 20150623 (Red Hat 4.8.5-36) (GCC)
`make CC=gcc FC=gfortran`

How to fix it? What other dependencies it need?

If I run `make install`, it can be finished successfully and show `Install OK!`.

But when I used the openblas functions by a demo, it would not work well.

The code
```c
#include <cblas.h>
#include <stdio.h>

int main()
{
  int i=0;
  double A[6] = {1.0,2.0,1.0,-3.0,4.0,-1.0};         
  double B[6] = {1.0,2.0,1.0,-3.0,4.0,-1.0};  
  double C[9] = {.5,.5,.5,.5,.5,.5,.5,.5,.5}; 
  cblas_dgemm(CblasColMajor, CblasNoTrans, CblasTrans,3,3,2,1,A, 3, B, 3,2,C,3);

  for(i=0; i<9; i++)
    printf("%lf ", C[i]);
  printf("\n");
  return 0;
}
```

compile
```shell
>$ gcc -I ~/OpenBLAS/include  -L ~/OpenBLAS/lib -lopenblas main.cpp -o solver
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `clagge_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `clatms_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `dlagsy_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `slatms_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `zlatms_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `slagsy_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `dlagge_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `clagsy_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `claghe_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `zlagsy_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `slagge_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `zlagge_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `dlatms_'
/es01/home/liyl/OpenBLAS/lib/libopenblas.so: undefined reference to `zlaghe_'
collect2: error: ld returned 1 exit status
```

--- a/lapack-netlib/TESTING/MATGEN/Makefile
+++ b/lapack-netlib/TESTING/MATGEN/Makefile
@@ -73,7 +73,7 @@ ZMATGEN = zlatms.o zlatme.o zlatmr.o zlatmt.o \
 endif
 
 .PHONY: all
-.NOTPARALLEL
+.NOTPARALLEL:
 all: $(TMGLIB)
 
 ALLOBJ = $(SMATGEN) $(CMATGEN) $(SCATGEN) $(DMATGEN) $(ZMATGEN) \

--- a/lapack-netlib/TESTING/MATGEN/Makefile
+++ b/lapack-netlib/TESTING/MATGEN/Makefile
@@ -40,27 +40,40 @@ ifneq ($(C_LAPACK), 1)
 	$(FC) $(FFLAGS) -c -o $@ $<
 endif
 
+ifneq "$(or $(BUILD_SINGLE),$(BUILD_COMPLEX))" ""
 SCATGEN = slatm1.o slatm7.o slaran.o slarnd.o
+endif
 
+ifeq ($(BUILD_SINGLE),1)
 SMATGEN = slatms.o slatme.o slatmr.o slatmt.o \
    slagge.o slagsy.o slakf2.o slarge.o slaror.o slarot.o slatm2.o \
    slatm3.o slatm5.o slatm6.o slahilb.o
+endif
 
+ifeq ($(BUILD_COMPLEX),1)
 CMATGEN = clatms.o clatme.o clatmr.o clatmt.o \
    clagge.o claghe.o clagsy.o clakf2.o clarge.o claror.o clarot.o \
    clatm1.o clarnd.o clatm2.o clatm3.o clatm5.o clatm6.o clahilb.o
+endif
 
+ifneq "$(or $(BUILD_DOUBLE),$(BUILD_COMPLEX16))" ""
 DZATGEN = dlatm1.o dlatm7.o dlaran.o dlarnd.o
+endif
 
+ifeq ($(BUILD_DOUBLE),1)
 DMATGEN = dlatms.o dlatme.o dlatmr.o dlatmt.o \
    dlagge.o dlagsy.o dlakf2.o dlarge.o dlaror.o dlarot.o dlatm2.o \
    dlatm3.o dlatm5.o dlatm6.o dlahilb.o
+endif
 
+ifeq ($(BUILD_COMPLEX16),1)
 ZMATGEN = zlatms.o zlatme.o zlatmr.o zlatmt.o \
    zlagge.o zlaghe.o zlagsy.o zlakf2.o zlarge.o zlaror.o zlarot.o \
    zlatm1.o zlarnd.o zlatm2.o zlatm3.o zlatm5.o zlatm6.o zlahilb.o
+endif
 
 .PHONY: all
+.NOTPARALLEL
 all: $(TMGLIB)
 
 ALLOBJ = $(SMATGEN) $(CMATGEN) $(SCATGEN) $(DMATGEN) $(ZMATGEN) \
@@ -107,9 +120,17 @@ cleanlib:
 	rm -f $(TMGLIB)
 
 ifneq ($(C_LAPACK), 1)
+ifeq ($(filter $(BUILD_SINGLE) $(BUILD_COMPLEX),1),)
 slaran.o: slaran.f ; $(FC) $(FFLAGS_NOOPT) -c -o $@ $<
+endif
+ifeq ($(filter $(BUILD_DOUBLE) $(BUILD_COMPLEX16),1),)
 dlaran.o: dlaran.f ; $(FC) $(FFLAGS_NOOPT) -c -o $@ $<
+endif
 else
+ifeq ($(filter $(BUILD_SINGLE) $(BUILD_COMPLEX),1),)
 slaran.o: slaran.c ; $(CC) $(CFLAGS) -O0 -c -o $@ $<
+endif
+ifeq ($(filter $(BUILD_DOUBLE) $(BUILD_COMPLEX16),1),)
 dlaran.o: dlaran.c ; $(CC) $(CFLAGS) -O0 -c -o $@ $<
+endif
 endif

